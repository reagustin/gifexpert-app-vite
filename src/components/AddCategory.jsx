import { useState } from 'react';
import PropTypes from 'prop-types';


export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    

    const onInputChange = ({target}) => {        
        setInputValue(target.value);        
    }
    
    const onSubmit = ( event ) => {        
        event.preventDefault();
        if (inputValue.trim().length <= 1) return;
        // setCategories( categories => [ inputValue, ...categories]);
        onNewCategory( inputValue.trim() );
        setInputValue('');
    }
// el onSubmit reacciona con mi enter, y envia el texto del input value.
  return (
    <form onSubmit={  onSubmit } aria-label="form">   
        <input
            type="text"
            placeholder="Buscar GIFs"
            value={inputValue}
            onChange={ onInputChange } //actualiza a medida que escribo en el textbox
        />
    </form>
  )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
}