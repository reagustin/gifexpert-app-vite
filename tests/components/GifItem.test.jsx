import { GifItem } from '../../src/components/GifItem'
import {render , screen} from '@testing-library/react';


describe('Pruebas en GifItem', () => { 

    const title = 'Saitama';
    const url = 'https://dragonball.com/goku.jpg'


    test('Debemos hacer match con un snapshot', () => { 
        
        const {container} =render( <GifItem title={title} url={url}  />);

        expect(container).toMatchSnapshot();

    });

    test('debe de mostrar la img con el url y el alt indicado', () => { 
        
        render( <GifItem title={title} url={url}  />);
        // screen.debug();

        expect(screen.getByRole('img').src).toBe(url);
        expect(screen.getByRole('img').alt).toBe(title);

    })

    test('debe de mostrar el titulo en el componente', () => { 
        render( <GifItem title={title} url={url}  />);
        expect(screen.getByText(title)).toBeTruthy();
    })

})
