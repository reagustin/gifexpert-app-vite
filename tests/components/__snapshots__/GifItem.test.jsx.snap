// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debemos hacer match con un snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://dragonball.com/goku.jpg"
    />
    <p>
       
      Saitama
       
    </p>
  </div>
</div>
`;
